docs/next/guides/accessibility/keyboard-shortcuts.md

```diff
@@ -0,0 +1,199 @@
+
+To manage keyboard shortcuts programmatically, you need to obtain the relevant context object from the API using the `getContext()` and execute one of its methods as explained below.
+
+### Removing keyboard shortcuts
+To remove an already registered keyboard shortcut (such as one of the default keyboard shortcuts), you need to search for it in the relevant context and refer to it by:
+
+- Either the key variant
+- Or the namespace
+
+Use the context's method `removeShortcutByVariants` to remove all shortcuts registered for given key variants (note that it is possible that there are multiple actions registered for a single keyboard shortcut):
+
+```js
+const gridContext = hot.getShortcutManager().getContext('grid');
+
+gridContext.removeShortcutByVariants([['enter']]);
+```
+
+Use the context's method `removeShortcutByNamespace` to remove all shortcuts registered in a certain namespace:
+
+```js
+const gridContext = hot.getShortcutManager().getContext('grid');
+
+gridContext.removeShortcutByNamespace('NAMESPACE_ID');
+```
+
+### Adding custom keyboard shortcuts
+
+Use the context's `addShortcut` method to register an action for a given keyboard shortcut.
+
+Within a single context, there might be multiple actions registered for the same keyboard shortcut. Your action will be simply added at the end of the stack of already defined actions.
+
+```js
+const gridContext = hot.getShortcutManager().getContext('grid');
+
+gridContext.addShortcut({ namespace: 'NAMESPACE_ID', variants: [['enter']], callback: () => {} });
+```
+
+If your action must run before a certain other action, you can refer to the other action by its namespace:
+
+```js
+const gridContext = hot.getShortcutManager().getContext('grid');
+
+gridContext.addShortcut({ namespace: 'NAMESPACE_ID', variants: [['enter']], callback: () => {}, position: 'before', relativeToNamespace: 'ANOTHER_NAMESPACE_ID' });
+```
+
+If your action must run only if some specific precondition is met, you can check for the precondition using a function provided to the `runAction` property:
+
+
+```js
+const gridContext = hot.getShortcutManager().getContext('grid');
+
+gridContext.addShortcut({ namespace: 'NAMESPACE_ID', variants: [['enter']], callback: () => {}, runAction: () => hot.getSelected() !== void 0 });
+```
+
+### Replacing default keyboard shortcuts
+To replace some keyboards shortcut's action by another one action you have to choose proper context, remove already registered shortcut and register another one.
+
+```js
+const gridContext = hot.getShortcutManager().getContext('grid');
+
+gridContext.removeShortcutByVariants(['enter']);
+gridContext.addShortcut({ namespace: 'NAMESPACE_ID', variants: [['enter']], callback: () => {} });
+```
+
+### Managing contexts
+
+Apart from the possibility of managing the built-in contexts listed above (`grid`, `editor`, `menu`), you are also free to create custom contexts.
```

Comment:
@wszymanski do you have something in mind why it might be useful to create custom contexts, as a developer who customizes Handsontable? If yes, can you add some explanation here?

Reply:
I think a good example of creating the new context would be a custom editor that after opening would show a big dialog box with a more sophisticated UI that requires for example the same arrow keys as for grid navigation. In that case, it would be much better to add shortcuts to the new context and make it active while playing with an editor. The same could be applied to the plugins that would have a more complex UI.

Reply:
Good point! 

Without going into full detail, we could say that `plugins/contextMenu/menu.js` is such an example. It uses `addContext` and `setActiveContextName` to manages its own context. This allows controlling all keyboard shortcuts within the plugin and returning the control to the grid.

However, one thing that needs explaining. How does the ContextMenu give control back to the `grid`, when the context menu is closed? I was expecting to find `setActiveContextName('grid')` somewhere in the `close()` method in `plugins/contextMenu/menu.js`, but it's not there. @wszymanski

Reply:
> However, one thing that needs explaining. How does the ContextMenu give control back to the `grid`, when the context menu is closed? I was expecting to find `setActiveContextName('grid')` somewhere in the `close()` method in `plugins/contextMenu/menu.js`, but it's not there. @wszymanski

It's related to fact that `ContextMenu` is a Handsontable in itself. We set context only for menu. The main Handsontable instance still has `grid` context. After closing context menu, we focus on the main HOT.